test-data/unit/check-literal.test

```diff
@@ -2643,3 +2643,306 @@ c_wrap: Literal[4, c]  # E: Invalid type "__main__.c" \
                        # E: Parameter 2 of Literal[...] is invalid
 [out]
+
+--
+-- Tests for Literals and enums
+--
+
+[case testLiteralWithEnumsBasic]
+from typing_extensions import Literal
+from enum import Enum
+
+class Color(Enum):
+    RED = 1
+    GREEN = 2
+    BLUE = 3
+
+    def func(self) -> int: pass
+
+r: Literal[Color.RED]
+g: Literal[Color.GREEN]
+b: Literal[Color.BLUE]
+bad: Literal[Color]   # E: Parameter 1 of Literal[...] is invalid
+
+def expects_color(x: Color) -> None: pass
+def expects_red(x: Literal[Color.RED]) -> None: pass
+def bad_func(x: Color.RED) -> None: pass             # E: Invalid type: try using Literal[__main__.Color.RED] instead?
+
+expects_color(r)
+expects_color(g)
+expects_color(b)
+expects_red(r)
+expects_red(g)  # E: Argument 1 to "expects_red" has incompatible type "Literal[__main__.Color.GREEN]"; expected "Literal[__main__.Color.RED]"
+expects_red(b)  # E: Argument 1 to "expects_red" has incompatible type "Literal[__main__.Color.BLUE]"; expected "Literal[__main__.Color.RED]"
+
+reveal_type(expects_red)  # E: Revealed type is 'def (x: Literal[__main__.Color.RED])'
+reveal_type(r)            # E: Revealed type is 'Literal[__main__.Color.RED]'
+reveal_type(r.func())     # E: Revealed type is 'builtins.int'
+[out]
+
+[case testLiteralWithEnumsBasicNewSemanticAnalyzer]
+# flags: --new-semantic-analyzer
+from typing_extensions import Literal
+from enum import Enum
+
+class Color(Enum):
+    RED = 1
+    GREEN = 2
+    BLUE = 3
+
+    def func(self) -> int: pass
+
+r: Literal[Color.RED]
+g: Literal[Color.GREEN]
+b: Literal[Color.BLUE]
+bad: Literal[Color]   # E: Parameter 1 of Literal[...] is invalid
+
+def expects_color(x: Color) -> None: pass
+def expects_red(x: Literal[Color.RED]) -> None: pass
+def bad_func(x: Color.RED) -> None: pass             # E: Invalid type: try using Literal[__main__.Color.RED] instead?
+
+expects_color(r)
+expects_color(g)
+expects_color(b)
+expects_red(r)
+expects_red(g)  # E: Argument 1 to "expects_red" has incompatible type "Literal[__main__.Color.GREEN]"; expected "Literal[__main__.Color.RED]"
+expects_red(b)  # E: Argument 1 to "expects_red" has incompatible type "Literal[__main__.Color.BLUE]"; expected "Literal[__main__.Color.RED]"
+
+reveal_type(expects_red)  # E: Revealed type is 'def (x: Literal[__main__.Color.RED])'
+reveal_type(r)            # E: Revealed type is 'Literal[__main__.Color.RED]'
+reveal_type(r.func())     # E: Revealed type is 'builtins.int'
+[out]
+
+[case testLiteralWithEnumsDefinedInClass]
+from typing_extensions import Literal
+from enum import Enum
+
+class Wrapper:
+    class Color(Enum):
+        RED = 1
+        GREEN = 2
+        BLUE = 3
+
+def foo(x: Literal[Wrapper.Color.RED]) -> None: pass
+
+r: Literal[Wrapper.Color.RED]
+g: Literal[Wrapper.Color.GREEN]
+
+foo(r)
+foo(g)  # E: Argument 1 to "foo" has incompatible type "Literal[__main__.Wrapper.Color.GREEN]"; expected "Literal[__main__.Wrapper.Color.RED]"
```

Comment:
Here I would however use `Wrapper.Color.RED`, so right thing would be probably to just skip the module name. I think this is already what we do for instances in cases like `x: int = str()`.